src/main/resources/archetype-resources/src/main/kotlin/__className__.kt

```diff
@@ -4,52 +4,56 @@ import java.util.logging.Logger
 
 import org.bonitasoft.engine.connector.AbstractConnector
 import org.bonitasoft.engine.connector.ConnectorValidationException
+import org.bonitasoft.engine.filter.AbstractUserFilter
+import org.bonitasoft.engine.api.APIAccessor
 
-class ${className} : AbstractConnector {
+class ${className} : AbstractUserFilter {
 
     constructor() : super()
 
-    val defaultInput = "defaultInput"
-    val defaultOutput = "defaultOutput"
+    companion object {
```

Comment:
Is the usage of a companion object mandatory ?

Reply:
Yes, it's the way to make a field 'static final' in kotlin. 
If we remove the companion then we cannot use 'const val' -> value will be on the instance and not the class. I did this for consistency with Java and Groovy.

Reply:
https://www.baeldung.com/kotlin-java-static-final